### ISSUE TITLE
Extension freezes when calling RPC method 

### ISSUE THREAD

**Issue by markdreyer** [MetaMask/metamask-extension#10119]
**Steps to reproduce (REQUIRED)**
1. Call RPC method "eth_GetEncryptionPublicKey" as in [docs](https://docs.metamask.io/guide/rpc-api.html#other-rpc-methods) 
2. Click on 'Provide' in popup titled "Request encryption public key"
3. Popup screen goes blank
4. Error in inspect of popup gives error connecting to sentry.io

Failed to load resource: the server responded with a status of 429 (Too Many Requests)
sentry.io/api/273496/store/?sentry_key=f59f3dd640d2429d9d0e2445a87ea8e1&sentry_version=7:1 
![image](https://user-images.githubusercontent.com/4984469/103078393-644bce80-4597-11eb-9dd0-1c399521dc69.png)
![image](https://user-images.githubusercontent.com/4984469/103078420-73cb1780-4597-11eb-9ffb-21015bb3a12b.png)

**Browser details **
 - OS: Windows
 - Browser: Brave Version 1.18.75 Chromium: 87.0.4280.101 (Official Build) (64-bit)

**Comment by markdreyer** [comment]
This could be a localhost issue. The error occurs when checking the origin at the code here:

https://github.com/MetaMask/metamask-extension/blob/ed19ce44bd83399e50ff4beba9118a2d591ad6e8/ui/app/pages/confirm-encryption-public-key/confirm-encryption-public-key.component.js#L162

**Comment by danfinlay** [comment]
Thanks for the report!

Do you have a `title` property assigned on the page you're doing this test on? I believe the `name` property is pulled off the requesting site's title. Aka, in html `<title>My Site</title>`.

**Comment by markdreyer** [comment]
Thanks! That was the issue. I was missing a `<title>` in my app. Now it gets further, but I get another similar error:
```
ui.js:formatted:23837 TypeError: Cannot read property 'from' of undefined
    at Function.mapToProps (ui.js:formatted:147013)
    at n (ui-libs.js:1)
    at h (ui-libs.js:1)
```
![image](https://user-images.githubusercontent.com/4984469/103113498-ad943080-4620-11eb-91b0-b2155fe54c3e.png)

It fully works in Chrome now, this is just an issue in Brave now. Perhaps it's customizations Brave has done or it's an older version. Should I open an issue on the Brave repo for that?

**Comment by markdreyer** [comment]
It's hard to tell from minified, but I think it's `confirmTransaction` being undefined in [confirm-decrypt-message.container.js](https://github.com/MetaMask/metamask-extension/blob/2ebf8756a4c1023e45e4bd98367f384836cb464a/ui/app/pages/confirm-decrypt-message/confirm-decrypt-message.container.js#L25)

**Comment by Gudahtt** [comment]
Thanks for the report! I believe this was fixed in #10180, which is included in v9.0.1 of the extension. We're in the process of rolling this out - it should be in Chrome within a few days hopefully.

I'm going to close this for now since the original error has been addressed, but this follow up error looks concerning as well :thinking:. If you see anything like that on MetaMask in any browser (i.e. not including Brave's MetaMask fork), I'd encourage you to open a new issue for it.
